### PR TITLE
Implement Webhooks

### PR DESCRIPTION
**Pull request by colindecarlo**
Changes:

* Remove unused Server, Subuser, User, EggVariable observers
* Remove unused Server, Subuser, and User events
*  Add WebhookConfiguration model to store where webhooks should be sent
* Add Webhook model to store previously executed webhooks
* Add Webhook Resource for Filament
* Add event new event listener (DispatchWebhooks) for eloquent model changes and our own app's events
* Fix weird issue with unused field_type for eggs

### CODE REVIEW

**Comment by Poseidon281** [comment]
I already made a PR for webhooks: [PR](https://github.com/pelican-dev/panel/pull/496)

**Comment by notAreYouScared** [comment]
If you wish to continue this, please open a new PR.

**Comment by lancepioch** [comment]
@notAreYouScared This is my PR.

**Comment by colindecarlo** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by go-vehikl** [comment]
I have read the CLA Document and I hereby sign the CLA

File: app/Traits/Services/HasWebhookPayload.php
Comment (lancepioch) [inline app/Traits/Services/HasWebhookPayload.php:1]: This should be removed as it's no longer used.

**Review by lancepioch — DISMISSED** [review]

**Review by notAreYouScared — APPROVED** [review]

**Pull request merged**